File: src/main/java/org/example/keycloak/hello/HelloResourceProvider.java
package org.example.keycloak.hello;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;
import java.util.Map;
import org.keycloak.models.KeycloakSession;
import org.keycloak.services.resource.RealmResourceProvider;

@Path("hello")
public class HelloResourceProvider implements RealmResourceProvider {

  private final KeycloakSession session;

  public HelloResourceProvider(KeycloakSession session) {
    this.session = session;
  }

  @Override
  public Object getResource() {
    return this;
  }

  @Override
  public void close() {
  }

  @GET
  @Produces(MediaType.APPLICATION_JSON)
  public Map<String, String> hello() {
    return Map.of("message", "Hello, World!");
  }
}

File: src/main/kotlin/org/example/keycloak/provider/HelloResourceProvider.kt
package org.example.keycloak.provider

import jakarta.ws.rs.GET
import jakarta.ws.rs.Produces
import jakarta.ws.rs.core.MediaType
import org.keycloak.models.KeycloakSession
import org.keycloak.services.resource.RealmResourceProvider

class HelloResourceProvider(private val session: KeycloakSession) : RealmResourceProvider {

    override fun getResource(): Any = this

    override fun close() {
        // No resources to close
    }

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun hello(): Map<String, String> = mapOf("message" to "Hello, World!")
}